src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -232,10 +223,68 @@ public bool TryGetWellKnownAttribute(WellKnownAttributeFlags flag, out bool valu
                 _name = "value";
             }
 
-            _packedFlags = new PackedFlags(refKind, hasByRefBeforeCustomModifiers, attributesAreComplete: handle.IsNil);
+            _packedFlags = new PackedFlags(refKind, attributesAreComplete: handle.IsNil);
 
             Debug.Assert(refKind == this.RefKind);
-            Debug.Assert(hasByRefBeforeCustomModifiers == this.HasByRefBeforeCustomModifiers);
+        }
+
+        private static PEParameterSymbol Create(
+            PEModuleSymbol moduleSymbol,
+            Symbol containingSymbol,
+            int ordinal,
+            bool isByRef,
+            ushort countOfCustomModifiersPrecedingByRef,
+            TypeSymbol type,
+            ParameterHandle handle,
+            ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers,
+            out bool isBad)
+        {
+            if (customModifiers.IsDefaultOrEmpty)
+            {
+                return new PEParameterSymbol(moduleSymbol, containingSymbol, ordinal, isByRef, type, handle, 0, out isBad);
+            }
+
+            return new PEParameterSymbolWithCustomModifiers(moduleSymbol, containingSymbol, ordinal, isByRef, countOfCustomModifiersPrecedingByRef, type, handle, customModifiers, out isBad);
+        }
+
+        private sealed class PEParameterSymbolWithCustomModifiers : PEParameterSymbol
+        {
+            private readonly ImmutableArray<CustomModifier> _customModifiers;
+            private readonly ushort _countOfCustomModifiersPrecedingByRef;
+
+            public PEParameterSymbolWithCustomModifiers(
+                PEModuleSymbol moduleSymbol,
+                Symbol containingSymbol,
+                int ordinal,
+                bool isByRef,
+                ushort countOfCustomModifiersPrecedingByRef,
+                TypeSymbol type,
+                ParameterHandle handle,
+                ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers,
+                out bool isBad) :
+                    base(moduleSymbol, containingSymbol, ordinal, isByRef, type, handle, customModifiers.Length, out isBad)
+            {
+                _customModifiers = CSharpCustomModifier.Convert(customModifiers);
+                _countOfCustomModifiersPrecedingByRef = countOfCustomModifiersPrecedingByRef;
+
```

Comment:
Would it make sense to assert here that `_countOfCustomModifiersPrecedingRef <= customModifiers.Length`?